gk/main.c

```diff
@@ -2215,8 +2215,11 @@ gk_proc(void *arg)
 				instance->has_insertion_failed = false;
 
 			iter_count = 0;
-		} else
-			iter_count++;
+		} else if (++iter_count == scan_iter) {
+			rte_prefetch_non_temporal(
+				&instance->ip_flow_entry_table[
+				(entry_idx + 1) % gk_conf->flow_ht_size]);
+		}
```

Comment:
There's so much in a single iteration of the `while` loop of `gk_proc()` that this prefetch could be "lost" by the time it's used. My suggestion to deal with this is to have an `if` statement before the first call to `send_pkts()` as follows:
```
if (iter_count >= scan_iter) {
	entry_idx = (entry_idx + 1) % gk_conf->flow_ht_size;
	rte_prefetch_non_temporal(&instance->ip_flow_entry_table[entry_idx]);
}
```
This way, the second `if (iter_count >= scan_iter) {` don't need to update `entry_idx` (reusing this computation) and the distance between the prefetch and its use is reduced.